### PR TITLE
Third party code from chromium and elsewhere.

### PR DESCRIPTION
**Pull request by phil-levis-google**
This pull request adds support code copied from chromium-c needed for the U2F application whose licenses cause it to fall under third party code. It also adds a `Makefile.app` so this code can be compiled as an external library for a libtock-c application. Some of the files have minor changes to reflect being built within libtock-c rather than a standalone firmware image. The changes are summarized here:

  - `sha256.c` adds `tock_compat_sha256` so that direct calls to SHA256 can use the digest hardware accelerator. Incremental calls with SHA256_INIT and SHA256_UPDATE still use the software implementation. It also removes a few helper #defines which are no longer needed.
  - `include/sha256.h` adds a declaration for `tock_compat_sha256` as well as HMAC/SHA256 system calls that are implemented in hardware (through system calls).
  - `include/timer.h` removes the dependency on task IDs since Tock doesn't use/need them.
  - `include/common.h` removes the dependencies on `config.h` and `module_id.h` since Tock doesn't use/need them. 
  - `include/signed_header.h` removes the BUILD_ASSERT calls that verify the size of `struct SignedHeader`.

### CODE REVIEW

File: third_party/chromiumos-ec/Makefile.app [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+CHROMIUM_OBJS   =  $(notdir $(patsubst %.c, %.o, $(CHROMIUM_CFILES)))
+
+define CHROMIUM_EC_RULES
+
+$$(BUILDDIR)/$(1)/chromiumos-ec:
+	$$(TRACEDIR)
+	$$(Q)mkdir -p $$@
+
+$(BUILDDIR)/$(1)/chromiumos-ec/sha256.o: $(CHROMIUMOS_DIR)/sha256.c | $$(BUILDDIR)/$(1)/chromiumos-ec
+#	echo $(BUILDDIR)
```
Comment (jrvanwhy) [inline third_party/chromiumos-ec/Makefile.app]: This file looks like it written for this PR (i.e. it's not pulled from chromium-ec). In that case, please remove the commented-out debug lines (14, 20, and 27)

File: third_party/chromiumos-ec/include/LICENSE [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2010 The Chromium OS Authors. All rights reserved.
```
Comment (jrvanwhy) [inline third_party/chromiumos-ec/include/LICENSE:1]: Please move this file one level higher in the directory structure (i.e. move it into chromium-ec/)

**Review by jrvanwhy — CHANGES_REQUESTED** [review]

File: third_party/chromiumos-ec/include/LICENSE (hunk 2 above)
Comment (phil-levis-google) [reply third_party/chromiumos-ec/include/LICENSE:1]: Is it OK that sha256.c (in the top level directory) has a different license?

File: third_party/chromiumos-ec/Makefile.app (hunk 1 above)
Comment (phil-levis-google) [reply third_party/chromiumos-ec/Makefile.app]: Done.

File: third_party/chromiumos-ec/include/LICENSE (hunk 2 above)
Comment (phil-levis-google) [reply third_party/chromiumos-ec/include/LICENSE:1]: Done.

Comment (jrvanwhy) [reply third_party/chromiumos-ec/include/LICENSE:1]: No, I guess we should leave it as-is.

Comment (phil-levis-google) [reply third_party/chromiumos-ec/include/LICENSE:1]: Reverted.

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**